index.js

```diff
@@ -282,36 +328,118 @@ class GhostInspector {
     return [data, passing]
   }
 
+  async waitForResult (resultId, options = { pollInterval: DEFAULT_POLL_INTERVAL }, callback) {
+    if (typeof options === 'function') {
+      callback = options
+      options = { pollInterval: DEFAULT_POLL_INTERVAL }
+    }
+    let result
+    try {
+      let passing = null
+      while (passing === null) {
+        await wait(options.pollInterval)
+        let pollFunction = this.getTestResult
+        if (options.resultType === 'suite') {
+          pollFunction = this.getSuiteResult
+        }
+        result = await pollFunction.call(this, resultId)
+        passing = result.passing
+      }
+    } catch (err) {
+      if (typeof callback === 'function') {
+        callback(err)
+        return
+      }
+      throw err
+    }
+    if (typeof callback === 'function') {
+      callback(null, result)
+    }
+    return result
+  }
+
+  async waitForTestResult (resultId, options, callback) {
+    return this.waitForResult(resultId, options, callback)
+  }
+
+  async waitForSuiteResult (suiteResultId, options, callback) {
+    if (typeof options === 'function') {
+      callback = options
+    }
+    options = options || {}
+    options.resultType = 'suite'
+    return this.waitForResult(suiteResultId, options, callback)
+  }
+
+  async executeTestOnDemand (organizationId, test, options, callback) {
+    assert.ok(test, 'test must be provided.')
+    if (typeof options === 'function') {
+      callback = options
+      options = {}
+    }
+    options = options || {}
+    options.body = test
+    let result
+    try {
+      result = await this.request('POST', `/organizations/${organizationId}/on-demand/execute/`, options)
+    } catch (err) {
+      if (typeof callback === 'function') {
+        callback(err)
+        return
+      }
+      throw err
+    }
+    if (options.wait) {
+      const pollInterval = options.pollInterval || DEFAULT_POLL_INTERVAL
```

Comment:
Does `|| DEFAULT_POLL_INTERVAL` need to be set since it defaults to that within the `waitForResult` method parameters anyway?

Reply:
The rest of the API allows (and checks) for `callback` in the options position. Since I explicitly pass `options` to the underlying `waitForResult()` function, this definition exists to maintain that pattern: if `options` is a function then specify `options` explicitly. Alternatively I could pass `undefined` in the `options` position?

Reply:
Yeah, looking at the code again, I guess the alternative is to pass `undefined` in the options position, I was trying to be explicit about it. Thoughts?

Reply:
I just realized I'm wiping out that `DEFAULT_POLL_INTERVAL` [here](https://github.com/ghost-inspector/node-ghost-inspector/pull/9/files#diff-168726dbe96b3ce427e7fedce31bb0bcR371) as well when I pass in the `options` object.

If I move that check for `options.pollInterval = options.pollInterval || DEFAULT_POLL_INTERVAL` into the `waitForResult()` method then I can enforce that there's actually a value in all places.

Reply:
I see what you mean. It's probably fine as is then.